src/agoradatatools/etl/transform/gene_info.py

```diff
@@ -5,6 +5,43 @@
 from agoradatatools.etl import transform
 
 
+def add_uniprot_id_to_gene_info(
+    gene_info: pd.DataFrame, uniprot_df: pd.DataFrame
+) -> pd.DataFrame:
+    """
+    This function will add a uniprotkb_accessions column containing uniprod IDs to the gene_info dataset.
+    The ensembl_gene_id is used to map the uniprot ID to the gene_info dataset.
+
+    Args:
+        gene_info (pd.DataFrame): The gene_info dataset to which the uniprot ID will be added.
+        uniprot_df (pd.DataFrame): The uniprot ID mapping dataset.
+
+    Returns:
+        pd.DataFrame: The gene_info dataset with the uniprot ID added.
+    """
+    # Check that the uniprot_df contains the expected columns
+    expected_columns = ["ensembl_gene_id", "uniprotkb_accessions"]
+    if not all(name in uniprot_df.columns for name in expected_columns):
+        raise ValueError(
+            f"uniprot_df does not contain the expected columns. Expected columns: {expected_columns}. Columns found: {uniprot_df.columns}"
+        )
+
+    # Collapse uniprot IDs into a list for each ensembl_gene_id
+    # This is necessary because there are multiple uniprot IDs for some ensembl_gene_id
+    collapsed_uniprot = (
+        uniprot_df.groupby("ensembl_gene_id")["uniprotkb_accessions"]
+        .apply(list)
+        .reset_index()
+    )
+
+    # Merge the datasets
+    gene_info = pd.merge(
+        left=gene_info, right=collapsed_uniprot, on="ensembl_gene_id", how="left"
+    )
+
+    return gene_info
+
+
 def transform_gene_info(
     datasets: dict, adjusted_p_value_threshold: float, protein_level_threshold: float
 ) -> pd.DataFrame:
```

Comment:
While you're editing this file, can you fix a typo I made in the function's main comment? It says "Each dataset will be left_joined onto gene_info", and it should be outer_joined.